### DISCUSSION TITLE
One user, multiple companies

### DISCUSSION THREAD

**Discussion by 2snEM6** [ory/kratos#2724] [Q&A] [answered]
Hi everyone! We are a SAAS company considering to migrate our custom made identity management system to Ory Kratos.

Currently, our model consists of a User entity (name, email, pwd)
which can be part of multiple Companies (Company entity).

Once the user authenticates, the user is then able to choose one of their companies. Our system then signs a JWT which contains the User Id and the Company Id. 

This JWT is sent back to the frontend, which includes
it on every API call back to the server.

I understand that Kratos does not support JWT but I read about the concept of Metadata.

Is there any way to replicate this behavior? Maybe include the company Id in the session metadata, but this should only be allowed to be done by the server (which is
the one that knows if a user is part of a company or not). Ideally the user could have multiple sessions, each one with a different company Id.

Thank you in advance.

**Comment by vinckr** [comment, marked as answer]
Hello @2snEM6 
Apologies for the late answer, maybe it still helps.

Ory Kratos can support your use case through the use of Identity Metadata. In Ory Kratos, there are two types of identity metadata: Public and Admin. Public metadata can be modified using the `/admin/identities` APIs and are visible to the user when calling `toSession()` or `/sessions/whoami`. Admin metadata can only be modified and read using the `/admin/identities APIs` and are never directly exposed to the `identity/user`.
In your case, you can store the Company Id in the Admin metadata. This ensures that only the server can modify it and it is not directly exposed to the user. You can manage metadata using the following APIs: `POST /admin/identities`, `GET /admin/identities`, `GET /admin/identities/{id}`, `PUT /admin/identities/{id}`. You will need an API Key to call these endpoints.

As for having multiple sessions for a user, each with a different Company Id, this is not directly supported by Ory Kratos. However, you can implement this behavior in your application by managing multiple sessions for a user and associating each session with a different Company Id in the Admin metadata.

Please note that metadata is not validated using the identity's JSON schema. You must ensure that the metadata you store is valid according to your schema and you must keep the schema up to date to accommodate changes in the data.

ps: marking this as answer due to the age of the discussion, but feel free to come back to this any time!
